pkg/event/target/mqtt.go

```diff
@@ -175,39 +181,34 @@ func NewMQTTTarget(id string, args MQTTArgs, doneCh <-chan struct{}, loggerOnce
 
 	client := mqtt.NewClient(options)
 
-	// The client should establish a first time connection.
-	// Connect() should be successful atleast once to publish events.
-	token := client.Connect()
-
 	target := &MQTTTarget{
 		id:         event.TargetID{ID: id, Name: "mqtt"},
 		args:       args,
 		client:     client,
 		loggerOnce: loggerOnce,
 	}
 
-	// Retries until the clientID gets registered.
-	retryRegister := func() {
-		// Repeat the pings until the client registers the clientId and receives a token.
+	go func(client mqtt.Client) {
```

Comment:
We could call this go func only if the queueDir is configured. Else we should return the error back as all the other targets do. Taking into account that, We should not try to set an invalid config - TestNotificationTargets() will not catch the connection errors if we are not returning them.

Reply:
we could do this in the else block ( if the queueDir is not configured )

```
token := client.Connect()
if token.Wait() && token.Error() != nil {
	return nil, token.Error()
}
```

Reply:
Or token.WaitTimeout

Reply:
in any case queueDir will be set in most production deployments, so we are invalidating the TestNotificationTargets here @Praveenrajmani - first, we shouldn't have two ways to do the same thing. It's confusing and wrong. 

It already made my testing harder that we have multiple ways to do target connect() so I would suggest we change every other target to do the same. Validation() of settings should happen regardless of queueDir is set or not. So I will keep this code as-is for now.

Reply:
The events will be lost if the queueDir is not set and target is inactive, And we silently ignore the invalid target and just keep logging the connection failures. Is this okay?

Reply:
yes why not @Praveenrajmani validation of the target is different than the initialization of the target for usage. If you wish to validate the args it should be the same for both queueDir set or not.

In the current master if queueDir is set there is no validation of the input at all and the connection retry is infinitely waiting which is a wrong implementation.